File: app/src/main/java/edu/uark/cartapp/models/api/ProductListing.java
package edu.uark.cartapp.models.api;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.LinkedList;
import java.util.List;

import edu.uark.cartapp.models.api.fields.ProductListingFieldName;
import edu.uark.cartapp.models.api.interfaces.LoadFromJsonInterface;

/* ==== APP ProductListing.java ====*/
public class ProductListing implements LoadFromJsonInterface<ProductListing> {
	private List<Product> products;

	public ProductListing() {
		this.products = new LinkedList<>();
	}

	public List<Product> getProducts() {
		return this.products;
	}
	public ProductListing setProducts(List<Product> products) {
		this.products = products;
		return this;
	}

	public ProductListing addProduct(Product product) {
		this.products.add(product);
		return this;
	}

	@Override
	public ProductListing loadFromJson(JSONObject rawJsonObject) {
		JSONArray jsonActivities = rawJsonObject.optJSONArray(ProductListingFieldName.PRODUCTS.getFieldName());

		if (jsonActivities != null) {
			try {
				for (int i = 0; i < jsonActivities.length(); i++) {
					JSONObject jsonActivity = jsonActivities.getJSONObject(i);
					this.products.add((new Product()).loadFromJson(jsonActivity));
				}
			} catch (JSONException e) {
				e.printStackTrace();
			}
		}
		return this;
	}
}

File: app/src/main/java/edu/uark/cartapp/models/api/ProductListing.kt
package edu.uark.cartapp.models.api

import edu.uark.cartapp.models.api.fields.ProductListingFieldName
import edu.uark.cartapp.models.api.interfaces.LoadFromJsonInterface
import org.json.JSONException
import org.json.JSONObject
import java.util.*

/* ==== APP ProductListing.java ====*/
class ProductListing : LoadFromJsonInterface<ProductListing> {
	private var products: MutableList<Product>? = null

	init {
		this.products = LinkedList()
	}

	fun getProducts(): List<Product>? {
		return this.products
	}

	fun setProducts(products: MutableList<Product>): ProductListing {
		this.products = products
		return this
	}

	fun addProduct(product: Product): ProductListing {
		this.products!!.add(product)
		return this
	}

	override fun loadFromJson(rawJsonObject: JSONObject): ProductListing {
		val jsonActivities = rawJsonObject.optJSONArray(ProductListingFieldName.PRODUCTS.fieldName)

		if (jsonActivities != null) {
			try {
				for (i in 0 until jsonActivities.length()) {
					val jsonActivity = jsonActivities.getJSONObject(i)
					this.products!!.add(Product().loadFromJson(jsonActivity))
				}
			} catch (e: JSONException) {
				e.printStackTrace()
			}
		}
		return this
	}
}